client/src/App.jsx

```diff
@@ -1,58 +1,70 @@
-import React, { Component } from 'react';
+import React, { Component } from "react";
 import {
-  BrowserRouter as Router, Route, Switch, Redirect,
-} from 'react-router-dom';
-import Login from './components/pages/Login';
-import TicketsPage from './components/pages/TicketsPage';
-import NewTicketPage from './components/pages/NewTicketPage';
-import ticketsSample from './components/model';
-import './App.css';
-import OpenedTicketPage from './components/pages/OpenedTicketPage';
-import SearchPage from './components/pages/SearchPage';
+  BrowserRouter as Router,
+  Route,
+  Switch,
+  Redirect
+} from "react-router-dom";
+import Login from "./components/pages/Login";
+import TicketsPage from "./components/pages/TicketsPage";
+import NewTicketPage from "./components/pages/NewTicketPage";
+import ticketsSample from "./components/model";
+import "./App.css";
+import OpenedTicketPage from "./components/pages/OpenedTicketPage";
+import SearchPage from "./components/pages/SearchPage";
+import socketIOClient from "socket.io-client";
+const socket = socketIOClient("http://localhost:7425");
 
 export default class App extends Component {
   state = {
     tickets: {
-      'all-tickets': {
+      "all-tickets": {
         pending: [],
-        closed: [],
+        closed: []
       },
-      'my-tickets': {
+      "my-tickets": {
         pending: [],
-        closed: [],
+        closed: []
       },
       drafts: [],
-      trash: [],
+      trash: []
     },
     search: {
-      query: '',
-      user: '',
-      status: '',
+      query: "",
+      user: "",
+      status: ""
     },
-    searchResults: null,
+    searchResults: null
   };
 
   componentDidMount() {
-    this.setState(ticketsSample);
+    socket.on("mails", data => {
+      this.setState({
+        tickets: [...this.state.tickets, JSON.parse(data)]
```

Comment:
It seems here that we receive the messages one by one. If so, it might be worth to wrap a promise around it and wait for all results to return and ```setState``` in the ```.then``` method. Otherwise we will be looking at a lot of unnecessary renders.

Reply:
but the main point here is to stream the emails one by one, so the user can see something and not just wait until all the messages are fetched